### ISSUE TITLE
Change filter parameter from "eq" to "cs"

### ISSUE THREAD

**Issue by chattago2002** [mevdschee/ra-data-treeql#6]
Hello.
Is there the possibility to use "cs" (string contains value) as filter parameter?
By default I noticed is set to "eq" but I can't find how to change it.

I'm referring to your REST API script (https://github.com/mevdschee/php-crud-api)

**Comment by mevdschee** [comment]
You need to change this line:

https://github.com/mevdschee/ra-data-treeql/blob/main/src/index.ts#L56

I hope this helps.

**Comment by chattago2002** [comment]
Sorry for my lacks, i'm starting using these tools from short time.
I can't find the index.ts file in my project.
After the command `npm install --save ra-data-treeql` I have this tree folder: "node_modules => ra-data-treeql=>dist" containing these files:
index.d.ts
index.js
index.js.map
metafile-cjs.json
metafile-esm.json
esm/index.js.map
esm/index.js

Looking for ",eq," in the "ra-data-treeql" folder, results are in:
index.js.map
esm/index.js
esm/index.js.map

Changing all occurrencies from ",eq," to ",cs," and restarting the project (in development) not help me.

Where I'm wrong?

**Comment by chattago2002** [comment]
I can't find the way to make this change.
The best way should be to set a parameter to select (for each component) which kind of "filter parameter" use.

For instance sometimes would be useful to have "cs" and others "eq" instead, but in my case the fastest and simplest solution would be to change all to "cs".

**Comment by nkappler** [comment]
Hi @chattago2002 

can you provide a few examples how these queries come to be?
I've also thought about changing this, so the search function would yield better results, but I am not sure if there are any negative impacts.

FYI the npm version is currently published from my fork of the repo...

**Comment by nkappler** [comment]
fixed via cf0b456d97ed1f25ed5dcbad15aaa98ab61d3cef

**Comment by chattago2002** [comment]
Hi @nkappler 
Thanks for you precious support and thanks also to @mevdschee for your repositories.

I'm using the latest version and I'm trying to use the implemented filter operators (with the suffix pattern) but probably I make some mistakes because I can see wrong calls when using filters. I'm following the [Filter Operators](https://marmelab.com/react-admin/FilteringTutorial.html#filter-operators) page from React-Admin documentation.

My filter is set up as following:
```
const filter = [<DateInput source="date_time_fld_sw" />];
```

The request payload is the following:
```
order: id,ASC
page: 1,100
filter: date_time_fld_sw,eq,2022-08-17
```

The source name is not "parsed" in "date_time_fld,sw". I tried others too, i.e. "cs" and "ge" but always the same wrong payload. Where am I wrong?

**Comment by nkappler** [comment]
Hi @chattago2002,
could you provide a codesandbox or could you check which parameters are passed to the data provider?

**Comment by chattago2002** [comment]
> could you check which parameters are passed to the data provider?

How can I do it?

**Comment by nkappler** [comment]
1. Open the Debugger (Firefox) / Sources (Chrome) Tab in the Browser's Developer Tools.
2. Open the `ra-data-treeql/index.ts` file with `Ctrl` + `P`
  In case you don't have source maps enabled, click on the initiator of the request in the Network Panel and find it from there:
  
![image](https://user-images.githubusercontent.com/9429273/185185373-21f069ca-db6b-4a53-ac72-af30c39e90d3.png)

3. Set a break point in the function sending the request, probably getList...
4. Copy the arguments passed to the function
![image](https://user-images.githubusercontent.com/9429273/185186809-1eb1b114-2730-4efb-928d-3258182a70ec.png)

When halted at the breakpoint, you should be able to copy the content of `params` into your clipboard in the context menu or by executing `copy(params)` in the console

**Comment by nkappler** [comment]
If you have written your own wrapper for the dataprovider, it is probably easier to set the breakpoint there.

**Comment by nkappler** [comment]
This is my minimal example and it seems to work flawlessly...

```tsx
import { Datagrid, List, TextField, TextInput } from "react-admin";

const filters = [
    <TextInput label="Search" source="q" alwaysOn />,
    <TextInput label="First Name" source="firstname_sw" defaultValue="" />,
];

export const CustomerList = () => (
    <List {...{ filters }}>
        <Datagrid rowClick="edit">
            <TextField source="firstname" />
            <TextField source="lastname" />
        </Datagrid>
    </List>
);
```

**Comment by nkappler** [comment]
![image](https://user-images.githubusercontent.com/9429273/185189258-73055f86-789c-4657-b190-42894af1d26f.png)
Request created by ra-data-treeql:
`http://localhost:3000/records/customer?order=id,ASC&page=1,10&filter=firstname,sw,Ma&search=Mu`

**Comment by nkappler** [comment]
It just appeared to me that react-admin is putting the filters into the url, maybe you can paste what's there for you.
My url changes like this when I start searching:
`http://localhost:3000/#/customers?displayedFilters=%7B%7D&filter=%7B%22q%22%3A%22Max%22%7D&order=ASC&page=1&perPage=10&sort=id`

**Comment by chattago2002** [comment]
> It just appeared to me that react-admin is putting the filters into the url

Yes, I noticed it too. Maybe in last version.

Anyway my problem with filters usage was about a partial update of your repository. When I went to its folder the version was 1.1.3 but probably something went wrong on update procedure. I removed and installed it again and all seems to work fine.
It's fantastic to use filter in this way. 😍
Thank you very much!!!
